### ISSUE TITLE
Adding items to favorites causes DIDL errors in subscription

### ISSUE THREAD

**Issue by tagdara** [SoCo/SoCo#516]
Last night my wife added an Amazon Music Playlist to Sonos Favorites and all of my Soco-based components stopped working (and logged 218k errors).

A quick dig through the logs returned the following error:
```
40   File "/usr/local/lib/python3.4/dist-packages/soco/data_structures.py", line 96, in from_didl_string
40     cls = _DIDL_CLASS_TO_CLASS[item_class]
40 KeyError: 'object.container.playlistContainer.sonos-favorite'
```

Previously I opened another issue to deal with trailing .# fields on otherwise known data types, and this one seems not much different.  I modified data_structures as follows, but I'm concerned that there are other ways this might break.   It also seems like there might be a better blanket way of resolving this - maybe catching the exception and trying the parent class rather than just falling over?  I believe this was considered previously and had other drawbacks since Sonos is so inelegant with datatypes.

```
def from_didl_string(string):
    """Convert a unicode xml string to a list of `DIDLObjects <DidlObject>`.

    Args:
        string (str): A unicode string containing an XML representation of one
            or more DIDL-Lite items (in the form  ``'<DIDL-Lite ...>
            ...</DIDL-Lite>'``)

    Returns:
        list: A list of one or more instances of `DidlObject` or a subclass
    """
    items = []
    root = XML.fromstring(string.encode('utf-8'))
    for elt in root:
        if elt.tag.endswith('item') or elt.tag.endswith('container'):
            item_class = elt.findtext(ns_tag('upnp', 'class'))

            # In case this class has an # specified unofficial
            # subclass, ignore it by stripping it from item_class
            if '.#' in item_class:
                item_class = item_class[:item_class.find('.#')]

            # stop sonos favorite tagged items from key errors
            if '.sonos-favorite' in item_class:
                item_class = item_class[:item_class.find('.sonos-favorite')]

            try:
                cls = _DIDL_CLASS_TO_CLASS[item_class]
            except KeyError:
                raise DIDLMetadataError("Unknown UPnP class: %s" % item_class)
            items.append(cls.from_element(elt))
        else:
            # <desc> elements are allowed as an immediate child of <DIDL-Lite>
            # according to the spec, but I have not seen one there in Sonos, so
            # we treat them as illegal. May need to fix this if this
            # causes problems.
            raise DIDLMetadataError("Illegal child of DIDL element: <%s>"
                                    % elt.tag)
    return items
```

**Comment by ghcs27** [comment]
The error simply means that our library is missing a Sonos Favorite class, as would be introduced by #478. Sadly, that PR has a problem with the import structure, so it might take a while until it is ready to merge.

**Comment by ghcs27** [comment]
I'll try to make a 'light' version of #478 without the problematic code to fix this.

**Comment by ghcs27** [comment]
@tagdara could you please check if this has been resolved by #478?
